### ISSUE TITLE
HelperText - dynamic variant demo should include a live region

### ISSUE THREAD

**Issue by thatblindgeye** [patternfly/patternfly-react#7263]
**Is this a new component or an extension of an existing one?**
What is the existing component, if any?
HelperText

**Describe the feature**
A clear and concise description of the new feature. What is the expected behavior?
Currently the [dynamic variant with static text demo](https://patternfly-react-pr-7250.surge.sh/components/helper-text/react-demos/#dynamic-variant-with-static-text) does not notify screen readers when helper text items update from an "error" to a "success", due to the icons having the `aria-hidden` attribute and the text itself not updating outside of styling.

Screen reader users should be notified when the status/variant of these helper text items get updated. Conversation regarding this feature can be see in PR #7250 . Depending on the implementation used, the a11y docs for this component may also need updating.

**Are there visuals for this feature? If applicable, please include examples for each state and for varying widths**
Include screenshots or links to Marvel or other mockups.
N/A

**Any other information?**
N/A
